design/pipeline-artifacts.md

```diff
@@ -33,20 +33,19 @@ Artifacts are distinct from other `resources` types, including `containers`, `re
 
 ---
 
-## Downloading artifacts: `downloadArtifact`
+## Downloading artifacts: `download`
 
-`downloadArtifact` is a shortcut for several tasks:
+`download` is a shortcut for two tasks:
```

Comment:
What is the experience when a pipeline does not have any artifacts published to it? Do both steps get added anyway? If so, should we combine the logic into a single step instead?

The nice thing about always showing the task, is that it advertises the feature. However it seems silly for a single-job pipeline.

Is it really only two tasks? Aren't artifacts extensible? What about when downloading artifacts from a pipeline with extensible artifact types published to it?

Reply:
Since we're limiting this syntax to only pipeline artifacts, I should probably change this to say that `download` is strictly a shortcut for the Pipeline Artifacts task. Users needing to publish artifacts to/download artifacts from file shares would directly reference the appropriate tasks, rather than using the `upload`/`download` shortcuts. Thoughts?

Reply:
This is not my understanding. It might be good to touch base for a few minutes later today. I've talked to multiple people and heard multiple answers.

Reply:
Context for my conversation with Alex: can one instance of `download` always resolve to a single task? I thought that's what we said in the PM spec, but apparently that's not what's being implemented right now.

Reply:
Yes, it should be a single task, as agreed.